src/ray/raylet/node_manager.cc

```diff
@@ -1904,18 +1904,22 @@ void NodeManager::ForwardTask(const Task &task, const ClientID &node_id,
 
   // Get and serialize the task's unforwarded, uncommitted lineage.
   auto uncommitted_lineage = lineage_cache_.GetUncommittedLineage(task_id, node_id);
-  Task &lineage_cache_entry_task =
-      uncommitted_lineage.GetEntryMutable(task_id)->TaskDataMutable();
+  auto entry = uncommitted_lineage.GetEntryMutable(task_id);
+  int num_forwards = -1;
+  if (entry) {
+    Task &lineage_cache_entry_task = entry->TaskDataMutable();
```

Comment:
You have to actually add the task to the uncommitted lineage if it is not already there. The receiving node manager expects the forwarded task to be in the lineage [here](https://github.com/ray-project/ray/blob/master/src/ray/raylet/node_manager.cc#L1015).

Reply:
Thanks, I guess it makes more sense to handle this all in this function.